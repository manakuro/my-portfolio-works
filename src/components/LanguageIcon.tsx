import * as React from 'react'
import { Language } from '@/modules/home/reducer'
import * as ReactTooltip from 'react-tooltip'

interface LanguageIconProps {
  language: Language
}

export const COMPONENTS = {
  vuejs: () => (
    <span className="icon-vuejs">
      <span className="path1" />
      <span className="path2" />
    </span>
  ),
  rails: () => (
    <span className="icon-vuejs">
      <span className="path1" />
      <span className="path2" />
    </span>
  ),
  react: () => (
    <span className="icon-vuejs">
      <span className="path1" />
      <span className="path2" />
    </span>
  ),
}

export default class LanguageIcon extends React.PureComponent<
  LanguageIconProps,
  {}
> {
  constructor(props: LanguageIconProps) {
    super(props)
  }

  public render(): JSX.Element {
    const Component = COMPONENTS[this.props.language.icon]
    if (!Component)
      throw new Error(
        `There is no such a component: ${this.props.language.icon}`,
      )

    return (
      <div className="tech-list-item" data-tip={this.props.language.name}>
        <Component />
        <ReactTooltip type="success" effect="solid" />
      </div>
    )
  }
}
