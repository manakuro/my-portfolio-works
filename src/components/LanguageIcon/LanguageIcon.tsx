import * as React from 'react'
import { Language } from '@/modules/home/reducer'
import * as ReactTooltip from 'react-tooltip'
import { onlyUpdateForKeys } from 'recompose'
import compose from 'recompose/compose'

export interface LanguageIconProps {
  language: Language
}

export const COMPONENTS = {
  vuejs: () => (
    <span className="icon-vuejs">
      <span className="path1" />
      <span className="path2" />
    </span>
  ),
  rails: () => (
    <span className="icon-vuejs">
      <span className="path1" />
      <span className="path2" />
    </span>
  ),
  react: () => (
    <span className="icon-vuejs">
      <span className="path1" />
      <span className="path2" />
    </span>
  ),
}

export const enhance = compose<LanguageIconProps, LanguageIconProps>(
  onlyUpdateForKeys(['language']),
)

export const LanguageIcon = (props: LanguageIconProps): JSX.Element => {
  const Component = COMPONENTS[props.language.icon]
  if (!Component)
    throw new Error(`There is no such a component: ${props.language.icon}`)

  return (
    <div className="language-icon" data-tip={props.language.name}>
      <Component />
      <ReactTooltip type="success" effect="solid" />
    </div>
  )
}

export default enhance(LanguageIcon)
