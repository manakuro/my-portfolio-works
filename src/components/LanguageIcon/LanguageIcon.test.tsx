import * as React from 'react'
import { shallow } from 'enzyme'
import * as LanguageIcon from './LanguageIcon'
import { LanguageIconProps } from '@/components/LanguageIcon/LanguageIcon'

const props: LanguageIconProps = {
  language: { id: 1, name: 'Ruby on Rails', icon: 'rails' },
}

describe('LanguageIcon', () => {
  it('should render', () => {
    const wrapper = shallow(<LanguageIcon.LanguageIcon {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
