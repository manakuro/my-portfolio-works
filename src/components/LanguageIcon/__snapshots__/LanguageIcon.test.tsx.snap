// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageIcon should render 1`] = `
<div
  className="language-icon"
  data-tip="Ruby on Rails"
>
  <rails />
  <ReactTooltip
    effect="solid"
    insecure={true}
    resizeHide={true}
    type="success"
    wrapper="div"
  />
</div>
`;
